MENTAL-MODEL.md

```diff
@@ -0,0 +1,376 @@
+# What is the Payjoin SDK and How does it work?
+
+The Payjoin SDK/`rust-payjoin` is the most well-tested and flexible library for BIP 78 Payjoin and related privacy practices.
+
+The primary crate, `payjoin`, is runtime-agnostic. Data persistence, chain interactions, and networking may be provided by custom implementations or copy the reference `payjoin-client` + bitcoind, `nolooking` + LND integration, or `bitmask-core` + BDK integrations.
+
+The following is a breakdown of the existing documentation and its application to the `payjoin-client` reference implementation.
+
+## Send a PayJoin
+
+The `sender` feature provides the check methods and PSBT data manipulation necessary to send payjoins. Just connect your wallet and an HTTP client. The reference implementation uses `reqwest` and Bitcoin Core RPC. Only a few non-default parameters are required:
+
+```rs
+fn send_payjoin(
+    bitcoind: bitcoincore_rpc::Client,
+    bip21: &str,
+    danger_accept_invalid_certs: bool,
+) -> Result<()>
+```
+
+Default modules including http and a bitcoin wallet may be useful additions to this library.
+
+The `danger_accept_invalid_certs` parameter is used for testing purposes only detailed in sectino 5.
+
+### 1. Parse BIP21 as `payjoin::Uri`
+
+Start by parsing a valid BIP 21 uri having the `pj` parameter. This is the [`bip21`](https://crates.io/crates/bip21) crate under the hood.
+
+```rs
+let link = payjoin::Uri::try_from(bip21)
+    .map_err(|e| anyhow!("Failed to create URI from BIP21: {}", e))?;
+
+let link = link
+    .check_pj_supported()
+    .map_err(|e| anyhow!("The provided URI doesn't support payjoin (BIP78): {}", e))?;
+```
```

Comment:
Refactoring ([see issue #19](https://github.com/payjoin/rust-payjoin/issues/19)) to use a builder pattern may eliminate the need to call `check_pj_supported` or expose bip21 at all.